docs/reference/creating_a_provider_class.rst

```diff
@@ -277,7 +277,6 @@ added to the provider pool.
                 - ['@sonata.media.resizer.simple']
         tags:
             - { name: sonata.media.provider }
-        public: true
```

Comment:
Hey @jordisala1991 

But the lest version of Media bundle support symfony 4.4 or higher

Reply:
Yes, but that does not relate to need a public service or not. In this case, we need it public because on SonataMediaBundle 3.x there are multiple places where we use the container directly to fetch services. That is not the case for 4.x. That's why there is this difference between the 2 versions.

Reply:
But in this case it's an example when I'll create a custom provider, I guess in this case the dev should create without public: true , don't you?

Reply:
This was the PR introducing the public explicitly for the providers: https://github.com/sonata-project/SonataMediaBundle/pull/1432

It doesn't explain why it is needed.
Anyway, on my opinion, it is safer to keep the public true here (unless you or someone else can confirm and searched on SonataMediaBundle code for any suspicious code that might get a provider directly from the container) and remove it from 4.x because it is already working without it.

Reply:
Ohhh now I got it

yeah you are right without `public:true`, it'll broken here

```
$this->container->get($media->getProviderName());
```

Reply:
Hey @jordisala1991 thank you for your explanation :) But I we are using provider without public:true and I don't get any kind of error 🤯 

```yaml
    sonata.media.provider.jpg:
        class: Sonata\MediaBundle\Provider\ImageProvider
        arguments:
            - sonata.media.provider.jpg
            - '@sonata.media.filesystem.local'
            - '@sonata.media.cdn.server'
            - '@sonata.media.generator.default'
            - '@sonata.media.thumbnail.format'
            - [ 'jpg' ]
            - [ 'image/pjpeg', 'image/jpeg' ]
            - '@sonata.media.adapter.image.gd'
            - '@sonata.media.metadata.proxy'
        tags:
            - { name: sonata.media.provider }
```